samples/paths.rb

```diff
@@ -1,4 +1,14 @@
-SAMPLES_PATH = Pathname.new(__dir__).relative_path_from(Pathname.pwd)
-CONFIG_PATH = SAMPLES_PATH.join('configuration')
-CLEAN_FILE = SAMPLES_PATH.join('clean.rb')
-SMELLY_FILE = SAMPLES_PATH.join('smelly.rb')
+SAMPLES_DIR = Pathname.new(__dir__).relative_path_from(Pathname.pwd)
+
+# Configuration paths
+CONFIGURATION_DIR = SAMPLES_DIR.join('configuration')
+REGULAR_CONFIGURATION_DIR = CONFIGURATION_DIR.join('regular_configuration')
+EMPTY_SUB_DIRECTORY_DIR = REGULAR_CONFIGURATION_DIR.join('empty_sub_directory')
+REGULAR_CONFIGURATION_FILE = REGULAR_CONFIGURATION_DIR.join('regular.reek.yml')
+HOME_DIR = CONFIGURATION_DIR.join('home')
+HOME_CONFIGURATION_FILE = HOME_DIR.join('home.reek.yml')
```

Comment:
Having more constants here makes the specs harder to follow I think.

Reply:
What are you suggesting as alternative?  For me having to parse 

```Ruby
SAMPLES_DIR.join('configuration').join('....')
```

potentially dozen of times in a spec file is even harder ;)

Reply:
Especially things like 

```Ruby
SAMPLES_DIR.join('.reek.yml')
```

are not expressive at all. What you really mean by that is the "configuration that you consider the right one if all things work out". But that line doesn't give you the slightest hint about that. We can argue about the naming, but at least REGULAR_CONFIGURATION_FILE seems a lot more helpful than that.

Reply:
> potentially dozen of times in a spec file is even harder ;)

How about moving these definitions to a `let` in the relevant spec file(s)?

> at least REGULAR_CONFIGURATION_FILE seems a lot more helpful than that.

I agree, but having it defined in some other file is less helpful. So having

```ruby
let(:regular_configuration_file) { SAMPLES_DIR.join('.reek.yml') }
```

at the top of the spec file would work for me.